uaclient/contract.py

```diff
@@ -65,9 +70,14 @@ def add_contract_machine(self, contract_token, machine_id=None):
 
         @return: Dict of the JSON response containing the machine-token.
         """
+        if not machine_id:
+            machine_id = system.get_machine_id(self.cfg)
+
         headers = self.headers()
         headers.update({"Authorization": "Bearer {}".format(contract_token)})
-        data = self._get_platform_data(machine_id)
+        activity_info = self._get_activity_info()
+        activity_info["lastAttachment"] = attachment_dt.isoformat()
```

Comment:
Won't this blow up when `attachment_dt` is `None`?

Reply:
yes, but I don't think attachment_dt is ever None (?) I can probably add types to make us more confident of that

Reply:
Oh I totally put a default arg value of `None` - that probably shouldn't be there